content/en/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions.md

```diff
@@ -178,6 +192,273 @@ Error from server (NotFound): Unable to list {"stable.example.com" "v1" "crontab
+
+```yaml
+type: object
+properties:
+  json:
+    x-kubernetes-preserve-unknown-fields: true
+```
+
+The field `json` can store any JSON value, without anything being pruned.
+
+It is possible to partially specify the permitted JSON, e.g.:
+
+```yaml
+type: object
+properties:
+  json:
+    x-kubernetes-preserve-unknown-fields: true
+    type: object
+    description: this is arbitrary JSON
+```
+
+With this only object type values are allowed.
+
+Pruning is enabled again for each specified property (or `additionalProperties`):
+
+```yaml
+type: object
+properties:
+  json:
+    x-kubernetes-preserve-unknown-fields: true
+    type: object
+    properties:
+      spec:
+        type: object
+        properties:
+          foo:
+            type: string
+          bar:
+            type: string
+```
+
+With this, the value:
+
+```yaml
+json:
+  spec:
+    foo: abc
+    bar: def
+    something: x
+  status:
+    something: x
+```
+
+will be pruned to:
+
+```yaml
+json:
+  spec:
+    foo: abc
+    bar: def
+  status:
+    something: x
+```
+
+This means that the `something` field in the specified `spec` object is pruned, but everything outside is not.
+
+### IntOrString
+
+Nodes in a schema with `x-kubernetes-int-or-string: true` are excluded from rule 1, such that the following is structural:
+
+```yaml
+type: object
+properties:
+  foo:
+    x-kubernetes-int-or-string: true
+``` 
+ 
+Also those nodes are partially excluded from rule 3 in the sense that the following two patterns are allowed (exactly those, without variations in order to additional fields):
+
+```yaml
+x-kubernetes-int-or-string: true
+anyOf:
+- type: integer
+- type: string
+...
+``` 
+
+and
+
+```yaml
+x-kubernetes-int-or-string: true
+allOf:
+- anyOf:
+  - type: integer
+  - type: string
+- ... # zero or more
+...
+``` 
+
+With one of those specification, both an integer and a string will validate.
+
+In [Validation Schema Publishing](/docs/tasks/access-kubernetes-api/extend-api-custom-resource-definitions/#publish-validation-schema-in-openapi-v2), `x-kubernetes-int-or-string: true` will be unfolded to one of the two patterns shown above.
+
+### RawExtension
+
+RawExtensions (as in `runtime.RawExtension` defined in [k8s.io/apimachinery](https://github.com/kubernetes/apimachinery/blob/03ac7a9ade429d715a1a46ceaa3724c18ebae54f/pkg/runtime/types.go#L94)) holds complete Kubernetes objects, i.e. with `apiVersion` and `kind` fields.
+
+It is possible specify those embedded objects (both completely without constraints or partially specified) by setting `x-kubernetes-embedded-object: true`. For example:
+
+```yaml
+type: object
+properties:
+  embedded:
```

Comment:
nit: somehow I had the first impression that the name `embedded` is special... can we rename it to something like `foo` or `embeddedFoo`?

Reply:
done